grumpy-runtime-src/runtime/dict.go

```diff
@@ -734,18 +841,23 @@ func dictPopItem(f *Frame, args Args, _ KWArgs) (item *Object, raised *BaseExcep
 	}
 	d := toDictUnsafe(args[0])
 	d.mutex.Lock(f)
-	iter := newDictEntryIterator(d)
-	entry := iter.next()
-	if entry == nil {
-		raised = f.RaiseType(KeyErrorType, "popitem(): dictionary is empty")
-	} else {
-		item = NewTuple(entry.key, entry.value).ToObject()
-		d.table.storeEntry(int(iter.index-1), deletedEntry)
-		d.table.incUsed(-1)
-		d.incVersion()
+	if d.table.used == 0 {
+		d.mutex.Unlock(f)
+		return nil, f.RaiseType(KeyErrorType, "popitem(): dictionary is empty")
```

Comment:
Again, what about `defer d.mutex.Unlock(f)` right after the Lock() call?

Reply:
As I think dict.popitem is quite useless method, ok I will.